url3aa01d34ec_katsuster_ememu_tgz-pJ8-net_katsuster_ememu_ui_MainJ8/src/net/katsuster/ememu/ui/MainApplet.java

```diff
@@ -0,0 +1,219 @@
+        System.out.println(proxyOpts);
+
+        //menu
+        listenButton = new ButtonListener();
+        setJMenuBar(new MainMenuBar(listenButton));
+
+        //tabs
+        tabPane = new JTabbedPane();
+        tabPane.setTabPlacement(JTabbedPane.BOTTOM);
+        tabPane.setFocusable(false);
+        tabPane.transferFocus();
+
+        //stdout Tab
+        panel = new JSplitPane();
+        panel.setDividerSize(4);
+
+        //stdout Tab - Left - stdout
+        stdoutPanel = new StdoutPanel(listenButton);
+        panel.setLeftComponent(stdoutPanel);
+
+        //stdout Tab - Right - Settings, Navigator
+        linuxOptPanel = new LinuxOptionPanel(linuxOpts);
+        proxyOptPanel = new ProxyOptionPanel(proxyOpts);
+
+        JPanel panelNavigator = new JPanel(new FlowLayout(FlowLayout.RIGHT));
+        JButton btnReset = new JButton("Reset");
+        btnReset.addActionListener(listenButton);
+        btnReset.setActionCommand("reset");
+        panelNavigator.add(btnReset);
+
+        JPanel panelRight = new JPanel(new GridLayout(3, 1, 5, 5), true);
+        panelRight.add(linuxOptPanel);
+        panelRight.add(proxyOptPanel);
+        panelRight.add(panelNavigator);
+        panelRight.setPreferredSize(new Dimension(100, 100));
+        panelRight.setMinimumSize(panelRight.getPreferredSize());
+        panel.setRightComponent(panelRight);
+
+        tabPane.addTab("stdout", panel);
+
+        setLayout(new BorderLayout());
+        add(tabPane);
+    }
+
+    @Override
+    public void start() {
+        System.out.println("start");
+
+        super.start();
+
+        //proxy
+        ProxyOption optProxy = proxyOptPanel.getOption();
+        System.setProperty("proxyHost", optProxy.getProxyHost().toString());
+        System.setProperty("proxyPort", Integer.toString(optProxy.getProxyPort()));
+
+        //stdout Tab - Left - stdout
+        stdoutPanel = new StdoutPanel(listenButton);
+        panel.setLeftComponent(stdoutPanel);
+
+        //terminal
+        for (int i = 0; i < vttyAMA.length; i++) {
+            if (vttyAMA[i] != null) {
+                tabPane.remove(vttyAMA[i]);
+                vttyAMA[i] = null;
+            }
+
+            vttyAMA[i] = new VirtualTerminal();
+            tabPane.addTab("ttyAMA" + i, vttyAMA[i]);
+        }
+        tabPane.setSelectedIndex(1);
+
+        //Run the emulator
+        emu = new Emulator();
+        emu.setOption(linuxOptPanel.getOption());
+        for (int i = 0; i < vttyAMA.length; i++) {
+            emu.getBoard().setUARTInputStream(i, vttyAMA[i].getInputStream());
+            emu.getBoard().setUARTOutputStream(i, vttyAMA[i].getOutputStream());
+        }
+        emu.start();
+    }
+
+    @Override
+    public void stop() {
+        System.out.println("stop");
+
+        super.stop();
+
+        try {
+            if (emu != null) {
+                emu.halt();
+                emu.join();
+            }
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)